src/traces/scattergl/convert.js

```diff
@@ -40,6 +41,46 @@ function convertStyle(gd, trace) {
 
     if(trace.visible !== true) return opts;
 
+    if(subTypes.hasText(trace)) {
+        opts.text = []
+
+        for (i = 0; i < trace.text.length; i++) {
+            var textOptions = opts.text[i] = {}
+
+            var textpos = unarr(trace.textposition, i)
+            textpos = textpos.split(/\s+/)
+            switch (textpos[1]) {
+                case 'left':
+                    textOptions.align = 'right';
+                    break;
+                case 'right':
+                    textOptions.align = 'left';
+                    break;
+                case 'center':
+                case 'centre':
```

Comment:
That's not a valid value. We should implement the same `textposition` values as scatter:

```
            'top left', 'top center', 'top right',
            'middle left', 'middle center', 'middle right',
            'bottom left', 'bottom center', 'bottom right'
```